src/bpmncwpverify/xml_ingest/bpmn_xml_ingestor.py

```diff
@@ -0,0 +1,106 @@
+from xml.etree.ElementTree import Element, parse
+from bpmncwpverify.bpmn.BPMN import (
+    Bpmn,
+    Task,
+    StartEvent,
+    EndEvent,
+    ExclusiveGatewayNode,
+    ParallelGatewayNode,
+    SequenceFlow,
+    Process,
+)
+
+
+################
+# Constants
+################
+from bpmncwpverify.constants import NAMESPACES
+
+TAG_CLASS_MAPPING = {
+    "task": Task,
+    "startEvent": StartEvent,
+    "endEvent": EndEvent,
+    "exclusiveGateway": ExclusiveGatewayNode,
+    "parallelGateway": ParallelGatewayNode,
+}
+
+FLOW_MAPPING = {"sequenceFlow": SequenceFlow}
+################
+
+
+# TODO: Add collaboration logic
+
+
+def _build_graph(process: Process) -> None:
+    for element_id, element_instance in process.elements.items():
+        for outgoing in element_instance.element.findall("bpmn:outgoing", NAMESPACES):
+            flow_id = outgoing.text
+            if not flow_id:
+                raise Exception("flow id is None")
+            flow = process.flows.get(flow_id.strip())
+            if flow is not None:
+                source_ref = flow.element.attrib["sourceRef"]
+                target_ref = flow.element.attrib["targetRef"]
+
+                # add target node id to neighbor list of current element
+                process.adj_list[element_id].append(target_ref)
+
+                # update flow's source_node
+                flow.source_node = process.elements[source_ref]
+                # update flow's target_node
+                flow.target_node = process.elements[target_ref]
+
+                # update source node's out flows array
+                process.elements[source_ref].out_flows.append(flow)
+                # update target node's in flows array
+                process.elements[target_ref].in_flows.append(flow)
+
+
+def _traverse_process(process_element: Element) -> Process:
+    process = Process(process_element)
+
+    for element in process_element:
+        tag_local = element.tag.partition("}")[2]
```

Comment:
Can `element.tag.partition("}")[2]` to a constant? Or a function? Give it a name too that reflects what it does.